src/gretel_trainer/relational/connectors.py

```diff
@@ -41,51 +42,58 @@ def __init__(self, engine: Engine):
             raise e
         logger.info("Successfully connected to db")
 
+    @classmethod
+    def from_conn_str(cls, conn_str: str) -> Connector:
+        """
+        Alternate constructor that creates a Connector instance
+        directly from a connection string.
+
+        Args:
+            conn_str: A full connection string for the target database.
+        """
+        engine = create_engine(conn_str)
+        return cls(engine)
+
     def extract(
         self,
         only: Optional[set[str]] = None,
         ignore: Optional[set[str]] = None,
         schema: Optional[str] = None,
+        config: Optional[ExtractorConfig] = None,
     ) -> RelationalData:
         """
         Extracts table data and relationships from the database. Optional args include:
-        - `only` (restrict extraction to these tables; cannot be used with `ignore`)
-        - `ignore` (exclude these tables from extraction; cannot be used with `only`)
-        - `schema` (limit scope to a specific schema; this is dialect-specific and not supported by all databases)
+
+        Args:
+            config: An optional extraction config. This config can be used to only include
+                specific tables, ignore specific tables, and configure subsetting. Please
+                see the `ExtractorConfig` docs for more details.
+
+        NOTE: The `only`, `ignore`, and `schema` args will be deprecated. Only the `config` param
+        will be used in the future, which contains all of the params from above.
         """
+        if only or ignore or schema:
+            logger.warning(
+                "Deprecation warning: in the future only the `config` param will be suported, please use an `ExtractorConfig` object for this param."
+            )
```

Comment:
Should we mark this entire `Connector#extract` method as deprecated and guide users towards using the `TableExtractor`? It seems like from an object responsibility perspective we're trying to simplify the Connector such that we'd eventually want users to basically implement a diff like the one below. Does this line up with what you're thinking?
```diff
connector = Connector.from_conn_string("...")
- rel_data = connector.extract()
+ config = ExtractorConfig(...)
+ extractor = TableExtractor(connector=connector, config=config, storage_dir=storage_dir)
+ extractor.sample_tables()  # *
+ rel_data = extractor.relational_data
```

\* I guess we'd currently need to set `refresh_relational_data=True` here, correct? Something about that feels strange but I'm not sure how I'd want to resolve it... have that value be the default? introduce some other function/method? something else?

Reply:
I think we should just change the whole UX up once we have scaling working better (working only w/ table refs on disk). Thinking about it more I might just remove `refresh_relational_data` as a param, and push that actual logic back into `Connector.extract` - it doesn't seem like a useful param directly to users and if we want to move away from loading DFs into memory then it's kinda pointless.

I can also just remove the existing deprecation warning, and just implicitly pass the `extract` kwargs to the `ExtractorConfig` - we can just have customers use a custom config if they need to subset sooner.